src/robot/devices/custom/rev-color-sensor-v3/index.ts

```diff
@@ -0,0 +1,289 @@
+        const buf = Buffer.alloc(1);
+        buf.writeUInt8(pulses);
+        await this._writeByte(Register.PROX_SENSOR_LED, freq | current);
+        await this._writeByte(Register.PROX_SENSOR_PULSES, buf.readUInt8());
+    }
+
+    /**
+     * Configure the proximity sensor
+     * @param res
+     * @param rate
+     */
+    public async configureProximitySensor(res: ProximitySensorResolution, rate: ProximitySensorMeasurementRate): Promise<void> {
+        await this._writeByte(Register.PROX_SENSOR_RATE, res | rate);
+    }
+
+    public async configureColorSensor(res: ColorSensorResolution, rate: ColorSensorMeasurementRate, gain: GainFactor): Promise<void> {
+        await this._writeByte(Register.LIGHT_SENSOR_MEAUSUREMENT_RATE, res | rate);
+        await this._writeByte(Register.LIGHT_SENSOR_GAIN, gain);
+    }
+
+    public getProximity(): number {
+        return this._lastProx;
+    }
+
+    public getRed(): number {
+        return this._lastRed;
+    }
+
+    public getGreen(): number {
+        return this._lastGreen;
+    }
+
+    public getBlue(): number {
+        return this._lastBlue;
+    }
+
+    public getIR(): number {
+        return this._lastIR;
+    }
+
+    public async hasReset(): Promise<boolean> {
+        const value = await this._readByte(Register.MAIN_STATUS);
+        return (value & 0x20) !== 0;
+    }
+
+    private async _checkDeviceID(): Promise<boolean> {
+        try {
+            const value = await this._readByte(Register.PART_ID);
+            if (value !== PART_IDENT) {
+                logger.error("Unknown device found with same I2C address, but incorrect ident");
+                return false;
+            }
+
+            return true;
+        }
+        catch (err) {
+            logger.error("Could not find REV Color Sensor");
+            return false;
+        }
+    }
+
+    private async _initializeDevice(): Promise<void> {
+        await this._writeByte(Register.MAIN_CTRL, MainControl.RGB_MODE | MainControl.LIGHT_SENSOR_ENABLE | MainControl.PROX_SENSOR_ENABLE);
+        await this._writeByte(Register.PROX_SENSOR_RATE, ProximitySensorResolution.PROX_RES_11BIT | ProximitySensorMeasurementRate.PROX_RATE_100MS);
+        await this._writeByte(Register.PROX_SENSOR_PULSES, 32);
+    }
+
+    private async _read11BitRegister(reg: Register): Promise<number> {
+        const buf = Buffer.alloc(2);
+        const offset = (reg as number);
+
+        buf[0] = await this._readByte(offset);
+        buf[1] = await this._readByte(offset + 1);
+
+        return buf.readUInt16LE() & 0x7FF;
+    }
+
+    private async _read20BitRegister(reg: Register): Promise<number> {
+        const buf = Buffer.alloc(4);
+        const offset = (reg as number);
+
+        buf[0] = await this._readByte(offset);
+        buf[1] = await this._readByte(offset + 1);
+        buf[2] = await this._readByte(offset + 2);
+
+        return buf.readUInt32LE() & 0x03FFFF;
```

Comment:
I actually copied this from REV's implementation, so I might leave it as is for now

Reply:
https://github.com/REVrobotics/Color-Sensor-v3/blob/61d9671f36ddfcb5581ae46ae5363089ebf3261b/src/main/java/com/revrobotics/ColorSensorV3.java#L439

Ok, since it's the way REV does it today, I agree, we should leave it. It is odd though.